crates/eframe/src/native/run.rs

```diff
@@ -88,53 +88,57 @@ impl<T: WinitApp> WinitAppWrapper<T> {
         event_loop: &ActiveEventLoop,
         event_result: Result<EventResult>,
     ) {
+        let mut now = Instant::now();
         let mut exit = false;
         let mut save = false;
 
         log::trace!("event_result: {event_result:?}");
 
-        let combined_result = event_result.and_then(|event_result| {
-            match event_result {
-                EventResult::Wait => {
-                    event_loop.set_control_flow(ControlFlow::Wait);
-                    Ok(event_result)
-                }
-                EventResult::RepaintNow(window_id) => {
-                    log::trace!("RepaintNow of {window_id:?}",);
+        let mut event_result = event_result;
 
-                    if cfg!(target_os = "windows") {
-                        // Fix flickering on Windows, see https://github.com/emilk/egui/pull/2280
-                        self.winit_app.run_ui_and_paint(event_loop, window_id)
-                    } else {
-                        // Fix for https://github.com/emilk/egui/issues/2425
-                        self.windows_next_repaint_times
-                            .insert(window_id, Instant::now());
-                        Ok(event_result)
-                    }
-                }
-                EventResult::RepaintNext(window_id) => {
-                    log::trace!("RepaintNext of {window_id:?}",);
+        if cfg!(target_os = "windows") {
+            if let Ok(EventResult::RepaintNow(window_id)) = event_result {
+                log::trace!("RepaintNow of {window_id:?}");
+                self.windows_next_repaint_times.insert(window_id, now);
+
+                // Fix flickering on Windows, see https://github.com/emilk/egui/pull/2280
+                event_result = self.winit_app.run_ui_and_paint(event_loop, window_id);
+
+                now = Instant::now();
+            }
+        }
+
+        let combined_result = event_result.map(|event_result| match event_result {
+            EventResult::Wait => {
+                event_loop.set_control_flow(ControlFlow::Wait);
+                event_result
+            }
+            EventResult::RepaintNow(window_id) => {
+                log::trace!("RepaintNow of {window_id:?}",);
+                self.windows_next_repaint_times.insert(window_id, now);
```

Comment:
please revert the change to `now` - it is only adding confusion for no apparent gain

```suggestion
                self.windows_next_repaint_times.insert(window_id, Instant::now());
```